### PR TITLE
Add forecasting demand components to azureml-assets.

### PR DESCRIPTION
**Pull request by nick863**
In this PR we are adding three sets of components used for forecasting demand.
- Single model inference component
- Many model training and inference components
- Hierarchical Time Series (HTS) training and inference components. 
The latter two are represented by two pipelines for training and inferencing respectively.

### CODE REVIEW

**Review by vlbejan — APPROVED** [review]

**Review by vlbejan — APPROVED** [review]

File: assets/training/forecasting_demand/environments/automl-gpu/environment.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+image:
+  name: azureml/curated/automl-gpu
```
Comment (vizhur) [inline assets/training/forecasting_demand/environments/automl-gpu/environment.yaml:2]: do you have MCR PR somewhere?

Comment (nick863) [reply assets/training/forecasting_demand/environments/automl-gpu/environment.yaml:2]: Where do we have this repository?

Comment (vlbejan) [reply assets/training/forecasting_demand/environments/automl-gpu/environment.yaml:2]: @vizhur we did not know anything about MCR PR. It was not mentioned in the component release doc. Do you have a complete doc/guideline to release components? Thanks.

Comment (nick863) [reply assets/training/forecasting_demand/environments/automl-gpu/environment.yaml:2]: I have created the PR to the MCR repository: https://github.com/microsoft/mcr/pull/2694

**Review by vlbejan — APPROVED** [review]

**Review by vizhur — APPROVED** [review]

**Pull request merged**